src/core/algorithms/dc/model/dc.cpp

```diff
@@ -0,0 +1,104 @@
+
+namespace algos::dc {
+
+bool DC::CheckAllEquality() const {
+    auto check = [](auto const& pred) {
+        return pred.GetOperator() == dc::OperatorType::kEqual and pred.IsCrossTuple() and
+               pred.IsOneColumn();
+    };
+
+    return std::all_of(predicates_.begin(), predicates_.end(), check);
+}
+
+bool DC::CheckOneInequality() const {
+    size_t count_eq = 0, count_ineq = 0;
+    for (Predicate const& pred : predicates_) {
+        if (pred.IsConstant()) return false;
+
+        Operator op = pred.GetOperator();
+        if (op == OperatorType::kEqual and pred.IsOneColumn()) {
+            count_eq++;
+        } else if (op != OperatorType::kEqual and op != OperatorType::kUnequal and
+                   pred.IsCrossTuple()) {
+            count_ineq++;
+        }
+    }
+
+    return count_eq + count_ineq == predicates_.size() && count_ineq == 1;
+}
+
+bool DC::CheckOneTuple() const {
+    ColumnOperand operand = predicates_.front().GetVariableOperand();
+    dc::Tuple tuple = operand.GetTuple();
+
+    auto check = [tuple](Predicate const& pred) {
+        return (pred.IsConstant() or pred.IsOneTuple()) and
+               pred.GetVariableOperand().GetTuple() == tuple;
+    };
+
+    return std::all_of(predicates_.begin(), predicates_.end(), check);
+}
+
+bool DC::CheckTwoTuples() const {
+    auto check = [](Predicate const& pred) { return pred.IsVariable() and pred.IsCrossTuple(); };
+
+    return std::all_of(predicates_.begin(), predicates_.end(), check);
+}
+
+DCType DC::GetType() const {
+    if (CheckAllEquality())
+        return DCType::kAllEquality;
+    else if (CheckOneInequality())
+        return DCType::kOneInequality;
+    else if (CheckOneTuple())
+        return DCType::kOneTuple;
+    else if (CheckTwoTuples())
+        return DCType::kTwoTuples;
+    else
+        return DCType::kMixed;
+}
+
+std::string DC::ToString() const {
+    if (predicates_.empty()) return {};
+
+    static constexpr char const* kNot = "!";
+    static constexpr char const* kAnd = " and ";
+
+    std::stringstream ss;
+    ss << kNot << '(' << predicates_.front().ToString();
+    for (auto pred = std::next(predicates_.begin()); pred != predicates_.end(); ++pred) {
+        ss << kAnd << pred->ToString();
+    }
+    ss << ')';
+
+    return ss.str();
+}
+
+void DC::ConvertEqualities() {
+    std::vector<dc::Predicate> res;
+    for (auto const& pred : predicates_) {
+        auto left = pred.GetLeftOperand();
+        auto right = pred.GetRightOperand();
+        if (pred.IsVariable() and pred.IsCrossColumn() and pred.IsCrossTuple() and
+            pred.GetOperator().GetType() == dc::OperatorType::kEqual) {
+            res.emplace_back(dc::OperatorType::kLessEqual, left, right);
+            res.emplace_back(dc::OperatorType::kGreaterEqual, left, right);
+        } else {
+            res.emplace_back(pred);
+        }
+    }
+
+    predicates_ = std::move(res);
+}
+
+}  // namespace algos::dc
```

Comment:
add a line break (in [test_input_data/TestDC5.csv](https://github.com/Desbordante/desbordante-core/pull/510/files#diff-b9d6044bade9ca7845eceef24327c0e9f365538e110d11742fd1e753ea404101) too)